polyglot/piranha/src/piranha.rs

```diff
@@ -63,21 +67,19 @@ impl SourceCodeUnit {
   // This is implements the main algorithm of piranha.
   fn _apply_rule(
     &mut self, rule: Rule, rules_store: &mut RuleStore, parser: &mut Parser,
-    scope_query: &Option<String>,
-  ) -> bool {
+    scope_query: &Option<String>, applied_edit_start: &Option<usize>
```

Comment:
Ensure that the next reported match occurs before (line-column number wise) the previously reported match.